### PR TITLE
ACTIN-1102 Scripts for sharing NKI reports

### PR DESCRIPTION
**Pull request by nedleitch**
The pipeline will no longer name the reports by the NKI hashed id, writing them instead under the ACTIN id to simplify our operational processes.

These new scripts will automate the process of copying the reports to the share bucket under the name expected by Kaiko, and also expose the ability to lookup the hashed id for a given ACTIN id.

I am aware that these do not utilise the "find..." functions, I am not sure they are warranted in this case as I do not think they will be used beyond NKI for the moment. That may come as part of a later update.

### CODE REVIEW

File: actin/operational/lookup_hash_for_actin-id [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Comment (pauldwolfe) [inline actin/operational/lookup_hash_for_actin-id:1]: Why put this in a separate file? Maybe just a function get you the same readability without needing to jump between files

File: actin/operational/upload_report_for_kaiko [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
```
Comment (pauldwolfe) [inline actin/operational/upload_report_for_kaiko:1]: what do you think about calling this `share_report_with_nki? I think the verb share is used elsewhere in this context and kaiko is more like the middle man.

**Review by pauldwolfe — COMMENTED** [review]
Soud

File: actin/operational/lookup_hash_for_actin-id (hunk 1 above)
Comment (kzuberihmf) [reply actin/operational/lookup_hash_for_actin-id:1]: also the mix of underscores and hyphen in the script name looks weird

Comment (nedleitch) [reply actin/operational/lookup_hash_for_actin-id:1]: > also the mix of underscores and hyphen in the script name looks weird

I didn't like either `actin_id` which fails to point out that an "ACTIN id" is a single thing, nor `actinid` which I find unreadable. I use this convention quite often and don't find it weird but I can see why you would say that.

In the interests of moving on I will use an underscore.

Comment (nedleitch) [reply actin/operational/lookup_hash_for_actin-id:1]: > Why put this in a separate file? Maybe just a function get you the same readability without needing to jump between files

Simon asked specifically for this capability.

Comment (kzuberihmf) [reply actin/operational/lookup_hash_for_actin-id]: > I didn't like either `actin_id` which fails to point out that an "ACTIN id" is a single thing, nor `actinid` which I find unreadable. I use this convention quite often and don't find it weird but I can see why you would say that.
> 
> In the interests of moving on I will use an underscore.

No worries if its a convention we are happy with

File: actin/operational/share_report_with_nki [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+LOOKUP_CMD="$(dirname "$0")/lookup_hash_for_actin_id"
+
+[[ $# -ne 1 ]] && echo "Provide the ACTIN id for the report" >&2 && exit 1
+
+report="${SOURCE_BUCKET}/report/${1}.actin.pdf"
+extended="${SOURCE_BUCKET}/extended/${1}.actin.extended.pdf"
+
+hash="$($LOOKUP_CMD "$1")"
+for source in "$report" "$extended"; do
+  echo $source
```
Comment (kzuberihmf) [inline actin/operational/share_report_with_nki:15]: minor: was this for debugging? in most of your echo's you add descriptive text.

**Review by kzuberihmf — APPROVED** [review]

File: actin/operational/share_report_with_nki (hunk 3 above)
Comment (nedleitch) [reply actin/operational/share_report_with_nki]: Thank-you.

**Pull request merged**